src/python/WMCore/MicroService/MSPileup/MSPileupData.py

```diff
@@ -0,0 +1,204 @@
+            docs.pop(key, None)
+    return docs
+
+
+def encodeTimestamps(doc):
+    """
+    Encode timestamps in MSPileup data document
+    """
+    for key in ["insertTime", "lastUpdateTime"]:
+        val = doc.get(key, None)
+        if val and isinstance(val, int):
+            # convert to string representation
+            doc[key] = time.strftime("%Y-%m-%dT%H:%M:%SZ", time.gmtime(val))
+    return doc
+
+
+def decodeTimestamps(doc):
+    """
+    Decode timestamps in MSPileup data document
+    """
+    for key in ["insertTime", "lastUpdateTime"]:
+        val = doc.get(key, None)
+        if val and isinstance(val, str):
+            # convert to seconds since epoch in GMT time zone
+            doc[key] = calendar.timegm(time.strptime(val, "%Y-%m-%dT%H:%M:%SZ"))
+    return doc
+
+
+class MSPileupData(MSCore):
+    """
+    MSPileupData class provide whole logic behind data used and stored by MSPileup module
+    """
+
+    def __init__(self, msConfig, **kwargs):
+        # python 2/3 super style
+        # super(MSPileupData, self).__init__(msConfig, **kwargs)
+        # python 3 super style
+        super().__init__(msConfig, **kwargs)
+
+        self.msConfig.setdefault("mongoDBRetryCount", 3)
+        self.msConfig.setdefault("mongoDBReplicaSet", None)
+        self.msConfig.setdefault("mongoDBPort", None)
+        self.msConfig.setdefault("mockMongoDB", False)
+
+        # A full set of valid database connection parameters can be found at:
+        # https://pymongo.readthedocs.io/en/stable/api/pymongo/mongo_client.html
+        mongoIndex = IndexModel('pileupName', unique=True)
+        collection = self.msConfig.get('mongoDBCollection', 'msPileupCollection')
+        msOutDBConfig = {
+            'database': self.msConfig['mongoDB'],
+            'server': self.msConfig['mongoDBServer'],
+            'replicaSet': self.msConfig['mongoDBReplicaSet'],
+            'port': self.msConfig['mongoDBPort'],
+            'username': self.msConfig['mongoDBUser'],
+            'password': self.msConfig['mongoDBPassword'],
+            'mockMongoDB': self.msConfig['mockMongoDB'],
+            'connect': True,
+            'directConnection': False,
+            'logger': self.logger,
+            'create': True,
+            'collections': [(collection, mongoIndex)]}
+        mongoDB = MongoDB(**msOutDBConfig)
+        self.msDB = getattr(mongoDB, self.msConfig['mongoDB'])
+        self.dbColl = self.msDB[collection]
+
+    def create(self, pdict):
+        """
+        Create and return pileup data from campaigns dictionary
+
+        :param pdict: a dictionary representing MSPileup data
+        :return: list of MSPileupError or empty list
+        """
+        # first, create MSPileupObj which will be validated against its schema
+        try:
+            obj = MSPileupObj(pdict)
+            doc = obj.data
+        except Exception as exp:  # pylint: disable=broad-except
+            msg = f"Failed to create MSPileupObj, {exp}"
```

Comment:
we might have to cast `exp` to a string.

Reply:
I could be wrong, but this error message would print only the exception name, not the content/message that comes with it. Hence the suggestion above. But you might want to check it before making any changes.